library/common/http/dispatcher.h

```diff
@@ -95,13 +93,19 @@ class Dispatcher : public Logger::Loggable<Logger::Id::http> {
     // An implementation option would be to have drainable header maps, or done callbacks.
     std::vector<HeaderMapPtr> metadata_;
     HeaderMapPtr trailers_;
+
+    std::atomic<bool> local_closed_{};
+    std::atomic<bool> remote_closed_{};
   };
 
   using DirectStreamPtr = std::unique_ptr<DirectStream>;
 
   // Everything in the below interface must only be accessed from the event_dispatcher's thread.
   // This allows us to generally avoid synchronization.
   DirectStream* getStream(envoy_stream_t stream_id);
+  void cleanup(DirectStream& stream_id);
+  void closeLocal(envoy_stream_t stream_id, bool end_stream);
+  void closeRemote(envoy_stream_t stream_id, bool end_stream);
```

Comment:
I think these should probably be private.

Reply:
they are